File: _프로그래머스/21-40/28. 카펫.py
"""
* Title: 카펫
https://school.programmers.co.kr/learn/courses/30/lessons/42842
"""
def solution(brown, yellow):
    store_x = 3
    store_y = 3

    for x in range(4, brown + 1):
        for y in range(3, x + 1):
            check_yellow = (x - 2) * (y - 2)
            if check_yellow == yellow and brown + yellow == x * y:
                store_x = x
                store_y = y
                break

    return [store_x, store_y]

File: _프로그래머스 js/28. 카펫.js
/*
* Title: 카펫
https://school.programmers.co.kr/learn/courses/30/lessons/428421111
*/
function solution(brown, yellow) {
  let ans_i = 3;
  let ans_j = 3;

  for (let i = yellow; i >= 1; i--) {
    if (yellow % i !== 0) {
      continue;
    }

    ans_i = i;
    ans_j = yellow / i;

    if ((ans_i + 2) * 2 + ans_j * 2 === brown) {
      break;
    }
  }

  return [ans_i + 2, ans_j + 2];
}

// [4, 3]
console.log(solution(10, 2));

// [5, 4]
console.log(solution(14, 6));
